### PR TITLE
Feat/text ts conversion

### PR DESCRIPTION
**Pull request by EdwardIrby**
convert to TS

### CODE REVIEW

File: packages/text/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap [hunk 1]
```diff
@@ -150,7 +150,6 @@ exports[`Storyshots Heading className override 1`] = `
       }
     }
     data-css-1uk9rqa=""
-    size="large"
```
Comment (danethurber) [inline packages/text/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap:153]: 👍

File: packages/text/src/css/index.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export default {
     textTransform: 'uppercase',
     fontSize: type.fontSizeXSmall,
     letterSpacing: type.letterSpacingXSmall,
-    lineHeight: type.lineHeighTight,
+    lineHeight: type.lineHeightTight,
```
Comment (danethurber) [inline packages/text/src/css/index.ts:50]: 👍

File: packages/text/src/react/__stories__/index.story.tsx [hunk 3]
```diff
@@ -18,27 +18,27 @@ const PaddingDecorator = storyFn => (
...
 const heading = storiesOf('Heading', module).addDecorator(PaddingDecorator)
 Object.keys(Heading.sizes).forEach(size =>
-  heading.add(`size: ${size}`, _ => (
-    <Heading size={size}>
+  heading.add(`size: ${size}`, () => (
+    <Heading size={size as keyof typeof Heading.sizes}>
       <h1>{size}</h1>
     </Heading>
   ))
 )
```
Comment (danethurber) [inline packages/text/src/react/__stories__/index.story.tsx:26]: i think this work if you do `Object.values` instead....maybe...and then you don't have to cast `size`

File: packages/text/src/react/__stories__/index.story.tsx [hunk 4]
```diff
@@ -18,27 +18,27 @@ const PaddingDecorator = storyFn => (
...
-heading.add('style override', _ => (
+heading.add('style override', () => (
   <Heading style={style}>
     <h2>pink</h2>
   </Heading>
 ))
-heading.add('className override', _ => (
-  <Heading className={className}>
+heading.add('className override', () => (
+  <Heading className={(className as unknown) as string}>
```
Comment (danethurber) [inline packages/text/src/react/__stories__/index.story.tsx:33]:
```suggestion
  <Heading className={...className}>
```

maybe? not sure

File: packages/text/src/react/code.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { compose, css } from 'glamor'
+import React, {HTMLAttributes} from 'react'
+import { useTheme, names } from '@pluralsight/ps-design-system-theme'
+
+import stylesheet from '../css'
+
+const style = ({ themeName } : { themeName: keyof typeof names; }) =>
```
Comment (danethurber) [inline packages/text/src/react/code.tsx:7]:
```suggestion
const style = ({ themeName } : { themeName: ValueOf<typeof names> }) =>
```

File: packages/text/src/react/code.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  compose(
+    css(stylesheet[`.psds-text__code`]),
+    css(stylesheet[`.psds-text__code.psds-theme--${themeName}`])
+  )
+
+const Code: React.FC<HTMLAttributes<HTMLElement>> = props => {
```
Comment (danethurber) [inline packages/text/src/react/code.tsx:13]: 👍 

i guess `code` isn't cool enough to get it's own typings

File: packages/text/src/react/heading.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import { compose, css } from 'glamor'
+import React, { HTMLAttributes } from 'react'
+import { useTheme, names } from '@pluralsight/ps-design-system-theme'
+
+import stylesheet from '../css'
+import * as vars from '../vars'
+
+const style = ({ themeName, size }: { themeName?: keyof typeof names; size?: keyof typeof vars.headingSizes }) =>
```
Comment (danethurber) [inline packages/text/src/react/heading.tsx:8]: these should use ValueOf i think

File: packages/text/src/react/heading.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      stylesheet[`.psds-text__heading--size-${size}.psds-theme--${themeName}`]
+    )
+  )
+
+interface HeadingStatics {
+  sizes: typeof vars.headingSizes
+}
+
+interface HeadingProps extends HTMLAttributes<HTMLHeadingElement> {
+  size?:  keyof typeof vars.headingSizes
```
Comment (danethurber) [inline packages/text/src/react/heading.tsx:23]: ValueOf here as well

File: packages/text/src/react/heading.tsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+const Heading: React.FC<HeadingProps> & HeadingStatics = ({children, size, ...props}) => {
+  const themeName = useTheme()
+
+  return React.cloneElement(React.Children.only(children as React.ReactElement), {
```
Comment (danethurber) [inline packages/text/src/react/heading.tsx:29]:
```suggestion
  if(!React.isValidElement(children)) return
  
  return React.cloneElement(React.Children.only(children), {
```

i think that should work....

File: packages/text/src/react/list/index.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+import React, {HTMLAttributes} from 'react'
+
+import { useTheme, names } from '@pluralsight/ps-design-system-theme'
+
+import ListItem from './list-item'
+
+import stylesheet from '../../css'
+import * as vars from '../../vars'
+
+const styles = ({ themeName, type}: { themeName: keyof typeof names; type: keyof typeof vars.listTypes}) =>
```
Comment (danethurber) [inline packages/text/src/react/list/index.tsx:11]: same ValueOf

File: packages/text/src/react/list/index.tsx [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+      css(stylesheet[`.psds-text__list--type-${type}`])
+    )
+
+interface ListStatics {
+  types: typeof vars.listTypes
+  Item: typeof ListItem
+}
+    
+interface ListProps extends HTMLAttributes<HTMLUListElement | HTMLOListElement> {
+  type?:  keyof typeof vars.listTypes
```
Comment (danethurber) [inline packages/text/src/react/list/index.tsx:24]: same ValueOf

File: packages/text/tsconfig.json [hunk 12]
```diff
@@ -3,6 +3,7 @@
   "compilerOptions": {
     "allowJs": true,
     "rootDir": null,
-    "jsx": "react"
+    "jsx": "react",
+    "strict": true
```
Comment (danethurber) [inline packages/text/tsconfig.json:7]: 💥

**Review by danethurber — CHANGES_REQUESTED** [review]

File: packages/text/src/react/list/index.tsx (hunk 11 above)
Comment (EdwardIrby) [reply packages/text/src/react/list/index.tsx:24]: Just realized this is exactly the same when I went back to look at the type. It also does not shorten it but I'll use it.

File: packages/text/src/react/__stories__/index.story.tsx (hunk 4 above)
Comment (EdwardIrby) [reply packages/text/src/react/__stories__/index.story.tsx]: Nope looks like this breaks it

**Review by danethurber — APPROVED** [review]

**Pull request merged**